File: v2/csharp/src/main/csharp/vestorly/Model/EventInput.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace vestorly.model {
  [DataContract]
  public class EventInput {
    
    
    [DataMember(Name="type", EmitDefaultValue=false)]
    public string type { get; set; }

    
    
    [DataMember(Name="referer", EmitDefaultValue=false)]
    public string referer { get; set; }

    
    
    [DataMember(Name="original_url", EmitDefaultValue=false)]
    public string originalUrl { get; set; }

    
    
    [DataMember(Name="originator_email", EmitDefaultValue=false)]
    public string originatorEmail { get; set; }

    
    
    [DataMember(Name="subject_email", EmitDefaultValue=false)]
    public string subjectEmail { get; set; }

    
    
    [DataMember(Name="parent_event_id", EmitDefaultValue=false)]
    public string parentEventId { get; set; }

    
    
    [DataMember(Name="originator_id", EmitDefaultValue=false)]
    public string originatorId { get; set; }

    
    
    [DataMember(Name="advisor_id", EmitDefaultValue=false)]
    public string advisorId { get; set; }

    
    
    [DataMember(Name="subject_id", EmitDefaultValue=false)]
    public string subjectId { get; set; }

    
    
    [DataMember(Name="event_content", EmitDefaultValue=false)]
    public EventContent eventContent { get; set; }

    
    
    [DataMember(Name="created_at", EmitDefaultValue=false)]
    public string createdAt { get; set; }

    
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class EventInput {\n");
      
      sb.Append("  type: ").Append(type).Append("\n");
      
      sb.Append("  referer: ").Append(referer).Append("\n");
      
      sb.Append("  originalUrl: ").Append(originalUrl).Append("\n");
      
      sb.Append("  originatorEmail: ").Append(originatorEmail).Append("\n");
      
      sb.Append("  subjectEmail: ").Append(subjectEmail).Append("\n");
      
      sb.Append("  parentEventId: ").Append(parentEventId).Append("\n");
      
      sb.Append("  originatorId: ").Append(originatorId).Append("\n");
      
      sb.Append("  advisorId: ").Append(advisorId).Append("\n");
      
      sb.Append("  subjectId: ").Append(subjectId).Append("\n");
      
      sb.Append("  eventContent: ").Append(eventContent).Append("\n");
      
      sb.Append("  createdAt: ").Append(createdAt).Append("\n");
      
      sb.Append("}\n");
      return sb.ToString();
    }
  }
  
  
}

File: v2/java/src/main/java/vestorly/client/model/EventInput.java
package vestorly.client.model;

import vestorly.client.model.EventContent;

import com.wordnik.swagger.annotations.*;
import com.fasterxml.jackson.annotation.JsonProperty;


@ApiModel(description = "")
public class EventInput  {
  
  public enum TypeEnum {
     page_view,  sign_up,  sign_in,  content_posted,  create_post,  publish_post,  update_post,  delete_post,  unpublish_post,  invite,  publish_newsletter,  publish_social,  click,  delivered,  open,  dropped,  bounce, 
  };
  private TypeEnum type = null;
  private String referer = null;
  private String originalUrl = null;
  private String originatorEmail = null;
  private String subjectEmail = null;
  private String parentEventId = null;
  private String originatorId = null;
  private String advisorId = null;
  private String subjectId = null;
  private EventContent eventContent = null;
  private String createdAt = null;

  
  /**
   **/
  @ApiModelProperty(required = true, value = "")
  @JsonProperty("type")
  public TypeEnum getType() {
    return type;
  }
  public void setType(TypeEnum type) {
    this.type = type;
  }

  
  /**
   **/
  @ApiModelProperty(value = "")
  @JsonProperty("referer")
  public String getReferer() {
    return referer;
  }
  public void setReferer(String referer) {
    this.referer = referer;
  }

  
  /**
   **/
  @ApiModelProperty(required = true, value = "")
  @JsonProperty("original_url")
  public String getOriginalUrl() {
    return originalUrl;
  }
  public void setOriginalUrl(String originalUrl) {
    this.originalUrl = originalUrl;
  }

  
  /**
   **/
  @ApiModelProperty(value = "")
  @JsonProperty("originator_email")
  public String getOriginatorEmail() {
    return originatorEmail;
  }
  public void setOriginatorEmail(String originatorEmail) {
    this.originatorEmail = originatorEmail;
  }

  
  /**
   **/
  @ApiModelProperty(required = true, value = "")
  @JsonProperty("subject_email")
  public String getSubjectEmail() {
    return subjectEmail;
  }
  public void setSubjectEmail(String subjectEmail) {
    this.subjectEmail = subjectEmail;
  }

  
  /**
   **/
  @ApiModelProperty(value = "")
  @JsonProperty("parent_event_id")
  public String getParentEventId() {
    return parentEventId;
  }
  public void setParentEventId(String parentEventId) {
    this.parentEventId = parentEventId;
  }

  
  /**
   **/
  @ApiModelProperty(value = "")
  @JsonProperty("originator_id")
  public String getOriginatorId() {
    return originatorId;
  }
  public void setOriginatorId(String originatorId) {
    this.originatorId = originatorId;
  }

  
  /**
   **/
  @ApiModelProperty(value = "")
  @JsonProperty("advisor_id")
  public String getAdvisorId() {
    return advisorId;
  }
  public void setAdvisorId(String advisorId) {
    this.advisorId = advisorId;
  }

  
  /**
   **/
  @ApiModelProperty(value = "")
  @JsonProperty("subject_id")
  public String getSubjectId() {
    return subjectId;
  }
  public void setSubjectId(String subjectId) {
    this.subjectId = subjectId;
  }

  
  /**
   **/
  @ApiModelProperty(value = "")
  @JsonProperty("event_content")
  public EventContent getEventContent() {
    return eventContent;
  }
  public void setEventContent(EventContent eventContent) {
    this.eventContent = eventContent;
  }

  
  /**
   **/
  @ApiModelProperty(value = "")
  @JsonProperty("created_at")
  public String getCreatedAt() {
    return createdAt;
  }
  public void setCreatedAt(String createdAt) {
    this.createdAt = createdAt;
  }

  

  @Override
  public String toString()  {
    StringBuilder sb = new StringBuilder();
    sb.append("class EventInput {\n");
    
    sb.append("  type: ").append(type).append("\n");
    sb.append("  referer: ").append(referer).append("\n");
    sb.append("  originalUrl: ").append(originalUrl).append("\n");
    sb.append("  originatorEmail: ").append(originatorEmail).append("\n");
    sb.append("  subjectEmail: ").append(subjectEmail).append("\n");
    sb.append("  parentEventId: ").append(parentEventId).append("\n");
    sb.append("  originatorId: ").append(originatorId).append("\n");
    sb.append("  advisorId: ").append(advisorId).append("\n");
    sb.append("  subjectId: ").append(subjectId).append("\n");
    sb.append("  eventContent: ").append(eventContent).append("\n");
    sb.append("  createdAt: ").append(createdAt).append("\n");
    sb.append("}\n");
    return sb.toString();
  }
}
